### PR TITLE
Add disabled mem control & improve memory efficiency

### PR DESCRIPTION
**Pull request by jt2594838**
Add DisabledMemController so the overhead of MemControl is testable.

### CODE REVIEW

**Comment by liukun4515** [comment]
maybe the error is caused by memory leak?

**Comment by jt2594838** [comment]
reuse BufferwriteProcessor and OverflowProcessor to improve memory efficiency and avoid memory leakage.

File: iotdb/src/main/java/org/apache/iotdb/db/engine/bufferwrite/BufferWriteProcessor.java [hunk 1]
```diff
@@ -454,38 +485,6 @@ public long getFileSize() {
...
-   */
-  public void rollToNewFile() {
-    // TODO : [MemControl] implement this
-  }
-
-  /**
-   * Check if this TsFile has too big metadata or file. If true, close current file and open a new
-   * one.
-   */
-  private boolean checkSize() {
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/bufferwrite/BufferWriteProcessor.java]: why you delete this function... 
I think this strategy is useful, and we need to consider where to call this method...

Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/bufferwrite/BufferWriteProcessor.java]: This is done in another place.

**Pull request merged**